app/models/access.rb

```diff
@@ -0,0 +1,38 @@
+class Access < ApplicationRecord
+  ALLOWED_RESOURCE_TYPES = %w[Organization FacilityGroup Facility].freeze
+
+  belongs_to :user, required: true
+  belongs_to :resource, polymorphic: true, optional: true
+
+  enum role: {
+    super_admin: "super_admin",
+    admin: "admin",
+    analyst: "analyst"
```

Comment:
Hmmmmm are these really roles still? Should we stick to `admin` and `analyst`, or go with, say, `view` and `manage`?

Reply:
Actually, on second thought, I think we should _keep_ this as roles! Here's something I really liked from [take 3](https://github.com/simpledotorg/simple-server/pull/1173/files#diff-a9804da7022a39203c40f4586796c96fR3-R7).

There's an explicit central spot where we translate which roles are allowed to manage which resources, and which roles are allowed to view which resources.

I think in this PR, we currently distribute that knowledge across several policies by selecting the appropriate `Access` scope to use (`accesses.admin` or just `accesses`). I think a more explicit capturing of this would be awesome.